ts/features/walletV3/payment/screens/WalletPaymentPickMethodScreen.tsx

```diff
@@ -19,76 +38,296 @@ import {
 import { walletPaymentPickPaymentMethod } from "../store/actions/orchestration";
 import {
   walletPaymentAllMethodsSelector,
+  walletPaymentAmountSelector,
   walletPaymentPickedPaymentMethodSelector,
   walletPaymentUserWalletsSelector
 } from "../store/selectors";
 
+// ----------------- SCREEN -----------------
+
 const WalletPaymentPickMethodScreen = () => {
   const dispatch = useIODispatch();
   const navigation = useNavigation<IOStackNavigationProp<AppParamsList>>();
+  const [shouldShowWarningBanner, setShouldShowWarningBanner] =
+    React.useState<boolean>(false);
+  const [selectedSavedMethod, setSelectedSavedMethod] = React.useState<
+    string | undefined
+  >(undefined);
+  const [selectedGenericMethod, setSelectedGenericMethod] = React.useState<
+    string | undefined
+  >(undefined);
```

Comment:
one kind has a `walletId`, the other has a `paymentMethodId`,  and since they cannot be mashed together the only possible way to have a single state would be by giving it a structure along the lines of 
```ts
{
kind:'saved'|'generic';
id: string;
}
```
after a bunch of playing around, I figured keeping two different states would be the best choice; what do you think?

Reply:
I think that updating two states when we can only mutate one for the same purpose is a waste of component renders. We could save both ids in the same state, we do not care much if it's a walletId or a methodId since we save only a wallet object after a filter. I think it's safe.